code/game/machinery/vending.dm

```diff
@@ -1320,3 +1316,128 @@
+*/
+
+/obj/machinery/vending/verb/tip(mob/user in view(1))
+	set name = "Tip over"
+	set category = "Object"
+	set src in view(1)
+	if(tippable = 0)
+		to_chat(user, "<span class='notice'>How do you propose to tip [src]?</span>")
+		return
+	if(anchored != 0)
+		to_chat(user, "<span class='notice'>[src] is firmly bolted to the floor</span>")
+		return
+	if(tipped != 0)
+		to_chat(user, "<span class='notice'>[src] is already tipped over.</span>")
+		return
+	else
+		src.visible_message("<b><font color=red>[user] starts to rock [src] back and forth!</font></b>")
+		do_after(user, 80, target = src)
+		src.icon_state = "liberationstation"
+		src.visible_message("<b><font color=red>[user] tips [src] to the ground with a thud!</font></b>")
+		climbable = !climbable
+		tipped = !tipped
+		pass_flags = LETPASSTHROW
+		return
+
+/obj/machinery/vending/verb/untip(mob/user in view(1))
+	set name = "Stand up"
+	set category = "Object"
+	set src in view(1)
+	src.visible_message("<b><font color=red>[user] starts to stand [src] back up!</font></b>")
+	do_after(user, 120, target = src)
+	src.icon_state = "[initial(icon_state)]"
+	src.visible_message("<b><font color=red>[user] stands [src] upright!</font></b>")
+	tipped = !tipped
+	climbable = !climbable
+	pass_flags = null
+	return
+
+/obj/machinery/vending/verb/climb_on()
+
+	set name = "Climb structure"
+	set desc = "Climbs onto a structure."
+	set category = null
+	set src in oview(1)
+
+	do_climb(usr)
+
+/obj/machinery/vending/MouseDrop_T(var/atom/movable/C, mob/user as mob)
+	if(..())
+		return
+	if(C == user)
+		do_climb(user)
+
+
+/obj/machinery/vending/proc/do_climb(var/mob/living/user)
+
+	if(!can_touch(user) || climbable != 1)
+		return
+
+	for(var/obj/O in range(0, src))
+		if(O.density == 1 && O != src && !istype(O, /obj/machinery/door/window)) //Ignores windoors, as those already block climbing, otherwise a windoor on the opposite side of a table would prevent climbing.
+			to_chat(user, "<span class='warning'>You cannot climb [src], as it is blocked by \a [O]!</span>")
+			return
+	for(var/turf/T in range(0, src))
+		if(T.density == 1)
+			to_chat(user, "<span class='warning'>You cannot climb [src], as it is blocked by \a [T]!</span>")
+			return
+	var/turf/T = src.loc
+	if(!T || !istype(T)) return
+
+	var/obj/machinery/door/poddoor/shutters/S = locate() in T.contents
+	if(S && S.density) return
+
+	usr.visible_message("<span class='warning'>[user] starts climbing onto \the [src]!</span>")
+	climber = user
+	if(!do_after(user, 50, target = src))
+		climber = null
+		return
+
+	if(!can_touch(user) || !climbable)
+		climber = null
+		return
+
+	S = locate() in T.contents
+	if(S && S.density)
+		climber = null
+		return
+
+	usr.loc = get_turf(src)
+	if(get_turf(user) == get_turf(src))
+		usr.visible_message("<span class='warning'>[user] climbs onto \the [src]!</span>")
+
+	climber = null
+
+/obj/machinery/vending/proc/can_touch(var/mob/user)

```

Comment:
Does this need an `ishuman()` check or are you okay with simple mobs etc doing it?